components/fonts/shaper.rs

```diff
@@ -414,12 +414,28 @@ impl Shaper {
                 })
             }
 
+            if servo_config::pref!(layout_variable_fonts_enabled) {
+                for (tag, value) in &options.variation_settings {
+                    let variations = &[hb_variation_t {
+                        tag: *tag,
+                        value: value.0,
+                    }];
+
+                    hb_font_set_variations(
+                        self.hb_font,
+                        variations.as_ptr(),
+                        variations.len() as u32,
+                    );
+                }
+            }
```

Comment:
This should be done in `Shaper::new` based on the variations from the font passed in. It doesn't need to be part of the `ShapingOptions`. In addition, you should set the variations all at once by getting a slice of the vector an passing it through the C FFI. I've done that here: https://github.com/mrobinson/servo/commit/d30d7d3ae5825ce58c83f014e4a32a4dabc98ef0